testing/test_junitxml.py

```diff
@@ -4,51 +4,60 @@
 from pathlib import Path
 from typing import cast
 from typing import List
+from typing import Optional
 from typing import Tuple
 from typing import TYPE_CHECKING
+from typing import TypeVar
+from typing import Union
 from xml.dom import minidom
 
-import py
 import xmlschema
 
 import pytest
 from _pytest.config import Config
 from _pytest.junitxml import bin_xml_escape
 from _pytest.junitxml import LogXML
+from _pytest.monkeypatch import MonkeyPatch
+from _pytest.pytester import Pytester
+from _pytest.pytester import RunResult
 from _pytest.reports import BaseReport
 from _pytest.reports import TestReport
 from _pytest.store import Store
 
+T = TypeVar("T")
+
 
 @pytest.fixture(scope="session")
-def schema():
+def schema() -> xmlschema.XMLSchema:
     """Return an xmlschema.XMLSchema object for the junit-10.xsd file."""
     fn = Path(__file__).parent / "example_scripts/junit-10.xsd"
     with fn.open() as f:
         return xmlschema.XMLSchema(f)
 
 
 @pytest.fixture
-def run_and_parse(testdir, schema):
+def run_and_parse(pytester: Pytester, schema: xmlschema.XMLSchema) -> T:
     """Fixture that returns a function that can be used to execute pytest and
     return the parsed ``DomNode`` of the root xml node.
 
     The ``family`` parameter is used to configure the ``junit_family`` of the written report.
     "xunit2" is also automatically validated against the schema.
     """
 
-    def run(*args, family="xunit1"):
+    def run(
+        *args: Union[str, "os.PathLike[str]"], family: Optional[str] = "xunit1",
+    ) -> Tuple[RunResult, "DomNode"]:
```

Comment:
would love to have some 👀 on type annotation here as I'm still learning

Reply:
LGTM!

Reply:
can `family` be `None`? -- otherwise it should just be `: str` -- I made a video on this topic: [python typing: Optional is not optional!](https://youtu.be/7XZYlDpSxOU?list=PLWBKAf81pmOaP9naRiNAqug6EBnkPakvY)

Reply:
ha, nice video @asottile (I click like 5 times) - I see what you put into "optional" meaning. But yes, `family` is optional *and* could be `None` as well 
https://github.com/pytest-dev/pytest/blob/9bdd7512db9844af81ac5c8601f8a58e3e92086d/testing/test_junitxml.py#L1305